logstash-core/src/main/java/org/logstash/execution/Filter.java

```diff
@@ -0,0 +1,69 @@
+package org.logstash.execution;
+
+import org.logstash.Event;
+
+/**
+ * A Filter is simply a mapping of {@link QueueReader} to a new {@link QueueReader}.
+ */
+public interface Filter extends AutoCloseable {
+
+    QueueReader filter(QueueReader reader);
+
+    void flush(boolean isShutdown);
+
+    @LogstashPlugin(name = "mutate")
+    final class Mutate implements Filter {
+
+        private final String field;
+
+        private final String value;
+
+        /**
+         * Required Constructor Signature only taking a {@link LsConfiguration}.
+         * @param configuration Logstash Configuration
+         * @param context Logstash Context
+         */
+        public Mutate(final LsConfiguration configuration, final LsContext context) {
+            this.field = configuration.getString("ls.plugin.mutate.field");
+            this.value = configuration.getString("ls.plugin.mutate.value");
+        }
+
+        @Override
+        public QueueReader filter(final QueueReader reader) {
+            return new QueueReader() {
+                @Override
+                public long poll(final Event event) {
+                    final long seq = reader.poll(event);
+                    if (seq > -1L) {
```

Comment:
When would the sequence be negative?

Reply:
@andrewvc see https://github.com/elastic/logstash/pull/9137/files#diff-cf259dc78531f006beca5e143b529554R13 ... `-1` indicates a failure to poll an event (think interrupts, timeouts).

Reply:
Ah, of course :)